src/components/VmSnapshots/NewSnapshotModal.jsx

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { Button, Modal, Icon, FormControl } from 'patternfly-react'
+import { msg } from '../../intl'
+
+class NewSnapshotModal extends Component {
+  constructor (props) {
+    super(props)
+    this.state = { showModal: false, description: '' }
+    this.open = this.open.bind(this)
+    this.close = this.close.bind(this)
+    this.handleSave = this.handleSave.bind(this)
+    this.handleDescriptionChange = this.handleDescriptionChange.bind(this)
+  }
+
+  open () {
+    this.setState({ showModal: true })
+  }
+
+  close () {
+    this.setState({ showModal: false, description: '' })
+  }
+
+  handleDescriptionChange (e) {
+    this.setState({ description: e.target.value })
+  }
+
+  handleSave () {
+    const snapshot = {
```

Comment:
It would be nice if the dialog can validate that the name is non-empty, non-blanks-only string.